sycl/include/CL/sycl/properties/accessor_properties.hpp

```diff
@@ -33,5 +35,82 @@ constexpr const auto &noinit =
 
 #endif
 
+namespace INTEL {
+namespace property {
+struct buffer_location {
+  template <int A> struct instance {
+    constexpr bool operator==(const buffer_location::instance<A> &) const {
+      return true;
+    }
+    constexpr bool operator!=(const buffer_location::instance<A> &) const {
+      return false;
+    }
+    template <int B>
+    constexpr bool operator==(const buffer_location::instance<B> &) const {
+      return false;
```

Comment:
Why not just return `A == B`? This will eliminate the overload with non-templated version of the function.